api.go

```diff
@@ -207,23 +207,14 @@ func (api *API) addResource(prototype jsonapi.MarshalIdentifier, source interfac
 	}
 
 	var ptrPrototype interface{}
-	var name string
 
 	if resourceType.Kind() == reflect.Struct {
 		ptrPrototype = reflect.New(resourceType).Interface()
-		name = resourceType.Name()
 	} else {
 		ptrPrototype = reflect.ValueOf(prototype).Interface()
-		name = resourceType.Elem().Name()
```

Comment:
save getting name from Go Struct Name if GetID doesn't return name

Reply:
saved this behaviour (get from struct if Identifier does not have name)